packages/api/cms-api/src/file-uploads/file-uploads.service.ts

```diff
@@ -38,4 +41,31 @@ export class FileUploadsService {
 
         return fileUpload;
     }
+
+    createHash(params: DownloadParams): string {
+        if (!this.config.download) {
+            throw new Error("File Uploads: Missing download configuration");
+        }
+
+        const hash = `file-upload:${params.id}:${params.timeout}`;
+
+        return createHmac("sha1", this.config.download.secret).update(hash).digest("hex");
+    }
+
+    createDownloadUrl(file: FileUpload, { relativeUrls = false }: { relativeUrls?: boolean } = {}): string {
+        if (!this.config.download) {
+            throw new Error("File Uploads: Missing download configuration");
+        }
+
+        const baseUrl = `${relativeUrls ? "" : this.config.download.apiUrl}/file-uploads`;
```

Comment:
is a absolute url really required? imho a relative url is enough.

especially if we rewrite the upload url using next (BFF)

Reply:
I like it because it's coherent with how the DAM behaves. I find it confusing if similar endpoints behave differently. So I would keep it unless you have a "strong objection" to this

Reply:
arguments against absolute url:
- it makes using the module more difficult, as you need to pass the API_URL into the module (that was the main reason for my comment)
- the API_URL might be wrong: if the url is proxied - which is not known by the api itself - a different domain needs to be used
    - example: downloads might be behind a CDN for performance reasons
    - example: downloads might be beind a BFF for security reasons (to not expose api)
    - this could be fixed by specifiying this domain in the config, which makes the config even more complicated

Imho the easiest solution is to omit this not-neccesarrily-known-by-api part and output a relative url only.

What is the reason why we have an absolute url in DAM, and don't we have exactly the same issues there as well?

Reply:
> What is the reason why we have an absolute url in DAM, and don't we have exactly the same issues there as well?

I suppose this was decided early on, where we didn't use Next in the site and haven't thought about using BFFs of the API yet.

IMO it should be the same for both the DAM and file uploads. If we decide to always use relative URLs, we should remove it for the DAM as well (breaking change).

Let's discuss this in our next meeting

Reply:
I've remove the support for absolute URLs: b6ef5af4960a3ca8af5f990de83a4ae5f7a77bca